services/checks/checks.go

```diff
@@ -94,32 +114,25 @@ func New(pmmVersion string) *Service {
 
 // Run runs checks service that grabs checks from Percona Checks service every interval until context is canceled.
 func (s *Service) Run(ctx context.Context) {
-	if f := os.Getenv(envCheckFile); f != "" {
-		s.l.Warnf("Using local test checks file: %s", f)
-
-		data, err := ioutil.ReadFile(f) //nolint:gosec
-		if err != nil {
-			s.l.Errorf("Failed to read test checks file: %s.", err)
-			return
-		}
-		checks, err := check.Parse(bytes.NewReader(data))
-		if err != nil {
-			s.l.Errorf("Failed to parse test checks file: %s.", err)
-			return
-		}
-
-		s.updateChecks(checks)
-		return
-	}
-
 	ticker := time.NewTicker(s.interval)
 	defer ticker.Stop()
 
+	go s.checkResults(ctx)
```

Comment:
I think we should do it synchronously/blocking while we can, and we can now. That will greatly simplify implementation.

Reply:
Done, check results until all results are ready or next checks run started (by timeout for now)